src/runtime_src/core/tools/xbutil2/OO_HostMem.cpp

```diff
@@ -0,0 +1,169 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_HostMem.h"
+#include "tools/common/XBUtilities.h"
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string.hpp>
+namespace po = boost::program_options;
+
+namespace {
+bool
+string2action(std::string str)
+{
+  std::transform(str.begin(), str.end(), str.begin(),
+                 [](auto c) { return static_cast<char>(std::tolower(c)); });
+
+  if (str == "enable")
+    return true;
+  else if (str == "disable")
+    return false;
+  else
+    throw xrt_core::generic_error(EINVAL, "Invalid host-mem action '" + str + "'");
+}
+
+void
+host_mem(xrt_core::device* device, bool action, uint64_t size)
+{
+  XBUtilities::sudo_or_throw("Root privileges required to enable host-mem");
+  device->set_cma(action, size); //can throw
+}
+
+} //end namespace 
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_HostMem::OO_HostMem( const std::string &_longName, bool _isHidden )
+    : OptionOptions(_longName, _isHidden, "Controls host-mem functionality")
+    , m_devices({})
+    , m_action("")
+    , m_size("")
+    , m_help(false)
+{
+  m_optionsDescription.add_options()
+    ("device,d", boost::program_options::value<decltype(m_devices)>(&m_devices)->multitoken(), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("action", boost::program_options::value<decltype(m_action)>(&m_action)->required(), "Action to perform: ENABLE or DISABLE")
+    ("size,s", boost::program_options::value<decltype(m_size)>(&m_size), "Size of host memory to be enabled (e.g. 256M, 1G)")
+    ("help,h", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+
+  m_positionalOptions.
+    add("action", 1 /* max_count */)
+  ;
+}
+
+void
+OO_HostMem::execute(const SubCmdOptions& _options) const
+{
+  XBUtilities::verbose("SubCommand option: Host Mem");
+
+  XBUtilities::verbose("Option(s):");
+  for (auto & aString : _options)
+    XBUtilities::verbose(std::string(" ") + aString);
+
+  // Honor help option first
+  if (std::find(_options.begin(), _options.end(), "--help") != _options.end()) {
+    printHelp();
+    return;
+  }
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(m_optionsDescription).positional(m_positionalOptions).run(), vm);
+    po::notify(vm); // Can throw
+  }
+  catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << "\n\n";
+
+    printHelp();
+    return;
```

Comment:
This is an error.  Since it is one, we wish to indicate so with the return command.
```suggestion
    throw xrt_core::error(std::errc::operation_canceled);
```

Reply:
fixed